contrib/aws/internal/eventbridge/eventbridge.go

```diff
@@ -0,0 +1,110 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package eventbridge
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/eventbridge"
+	"github.com/aws/aws-sdk-go-v2/service/eventbridge/types"
+	"github.com/aws/smithy-go/middleware"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+	"strconv"
+	"time"
+)
+
+const (
+	datadogKey      = "_datadog"
+	startTimeKey    = "x-datadog-start-time"
+	resourceNameKey = "x-datadog-resource-name"
+	maxSizeBytes    = 256 * 1024 // 256 KB
+)
+
+type messageCarrier map[string]string
+
+func (carrier messageCarrier) Set(key, val string) {
+	carrier[key] = val
+}
+
+func EnrichOperation(ctx context.Context, in middleware.InitializeInput, operation string) {
+	switch operation {
+	case "PutEvents":
+		handlePutEvents(ctx, in)
+	}
+}
+
+func handlePutEvents(ctx context.Context, in middleware.InitializeInput) {
+	params, ok := in.Parameters.(*eventbridge.PutEventsInput)
+	if !ok {
+		log.Debug("Unable to read PutEvents params")
+		return
+	}
+
+	for i := range params.Entries {
+		injectTraceContext(ctx, &params.Entries[i])
+	}
```

Comment:
nit

I think you can just do

```go
for _, entry := range params.Entries {
  injectTraceContext(ctx, entry)
}
```

Reply:
We actually can't, otherwise we modify a copy rather than the actual entry

Reply:
Ah, okay, I see what's going on now. Thanks for explaining.